gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -51,7 +52,7 @@
  */
 public class SourceState extends State {
 
-  private static final Set<Extract> EXTRACT_SET = Sets.newConcurrentHashSet();
+  private static final Set<Extract> EXTRACT_SET = Sets.newSetFromMap(new ConcurrentHashMap<Extract, Boolean>());
```

Comment:
The prior causes me so many headaches because it is only in guava 15+, but the hadoop dependencies use an older version of guava.